### PR TITLE
Use `vtk_encode_string` to embed logo file

### PR DESCRIPTION
**Pull request by Meakk**

### CODE REVIEW

**Comment by mwestphal** [comment]
>  It results in a smaller buffer, thus a smaller libf3d binary, and we will be able to use this feature later for any file, not only the images.

I'm a bit confused. Where is the image stored then  ?

**Comment by Meakk** [comment]
> I'm a bit confused. Where is the image stored then ?

Instead of storing the raw buffer, the PNG buffer is stored.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -1,24 +1,16 @@
 # Generate F3DIcon.h
 add_executable(embed embed.cxx)
-target_link_libraries(embed PUBLIC VTK::IOImage)
+target_link_libraries(embed PUBLIC VTK::IOCore)
 
-# On Windows, add VTK IOImage library to PATH
+# On Windows, add VTK IOCore library to WORKING_DIRECTORY
```
Comment (mwestphal) [inline src/CMakeLists.txt:5]: not only on windows

File: src/embed.cxx [hunk 2]
```diff
@@ -1,42 +1,36 @@
...
 
-  outfile << "const unsigned char F3DIconBuffer[] = {\n";
-  unsigned char* p = static_cast<unsigned char*>(icon->GetScalarPointer(0, 0, 0));
+  unsigned long length = vtkBase64Utilities::Encode(input.data(), input.size(), output.data());
 
-  for (int i = 0; i < dim[0] * dim[1] * nbComp; i++)
+  if (length != expectedLength)
   {
-    outfile << std::hex << "  0x" << static_cast<unsigned int>(p[i]) << ",\n";
+    std::cerr << "Base64 length is incorrect" <<std::endl;
```
Comment (mwestphal) [inline src/embed.cxx:23]: space after <<

**Review by mwestphal — APPROVED** [review]

**Pull request merged**